### PR TITLE
chore(build): remove tier4_autoware_utils.hpp evaluator/ simulator/

### PR DESCRIPTION
**Pull request by soblin**
## Description

https://github.com/autowarefoundation/autoware.universe/issues/4821

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [X] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [X] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### Summary by CodeRabbit

Release Notes:
- Refactor: Removed the inclusion of `tier4_autoware_utils/tier4_autoware_utils.hpp` header file and replaced it with specific headers in multiple files related to kinematic, localization, planning, and metrics calculations. This change improves modularity and reduces unnecessary dependencies.
- Bug Fix: Replaced the inclusion of `tier4_autoware_utils/tier4_autoware_utils.hpp` with appropriate headers in several files, ensuring correct functionality and compatibility with the `tier4_autoware_utils` library.
- Documentation: Updated the include statements in various files to reflect the changes made for better code organization and maintainability.

### CODE REVIEW

**Review by takayuki5168 — APPROVED** [review]

File: simulator/dummy_perception_publisher/src/node.cpp [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 #include "dummy_perception_publisher/node.hpp"
 
-#include "tier4_autoware_utils/tier4_autoware_utils.hpp"
+#include <tier4_autoware_utils/geometry/geometry.hpp>
```
Comment (maxime-clem) [inline simulator/dummy_perception_publisher/src/node.cpp:17]: [ Feel free to ignore this comment ]
Here the quotation marks `"` are changed to chevrons `<`.
I think the rule should be the following:
- `"..."` for local includes (current package).
- `<...>` for library and system includes (different packages, `std`, third-party libraries like `Eigen`, ...).

So here this change is good but it should be made consistent in all files if possible.

**Review by maxime-clem — APPROVED** [review]

File: simulator/dummy_perception_publisher/src/node.cpp (hunk 1 above)
Comment (soblin) [reply simulator/dummy_perception_publisher/src/node.cpp:17]: That's right, I'll double check other PRs as well

**Review by kminoda — APPROVED** [review]
LGTM

**Comment by soblin** [comment]
@yukkysaito @tkimura4 @TakaHoribe Can you approve this PR ?

**Review by TakaHoribe — APPROVED** [review]

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**